azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodTemplate.java

```diff
@@ -0,0 +1,140 @@
+   * @throws IOException If unable to read the template file.
+   */
+  private AzKubernetesV1PodTemplate(String templatePath) throws IOException {
+    File templateFile = Paths.get(templatePath).toFile();
+    this.podFromTemplate = (V1Pod) Yaml.load(templateFile);
+  }
+
+  /**
+   * @param templatePath Path where the template file is located.
+   * @return Singleton instance of this class.
+   * @throws IOException If unable to read the template file.
+   */
+  public static synchronized AzKubernetesV1PodTemplate getInstance(String templatePath)
+      throws IOException {
+    if (null == instance) {
+      instance = new AzKubernetesV1PodTemplate(templatePath);
+    }
+    return instance;
+  }
+
+  /**
+   * @return the {@link V1Pod} POD generated from the template.
+   */
+  public V1Pod getPodFromTemplate() {
+    return this.podFromTemplate;
+  }
+
+  /**
+   * @param filterPredicate Predicate to filter the init containers.
+   * @return The list of filtered init Containers derived from the POD specified in the template.
+   */
+  public List<V1Container> getInitContainers(Predicate<? super V1Container> filterPredicate) {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    if (null == spec) {
+      return Collections.emptyList();
+    }
+    List<V1Container> initContainers = spec.getInitContainers();
+    return initContainers == null ? Collections.emptyList() :
+        initContainers.stream().filter(filterPredicate).collect(Collectors.toList());
+  }
+
+  /**
+   * @return The list of init Containers derived from the POD specified in the template.
+   */
+  public List<V1Container> getInitContainers() {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    return spec == null ? Collections.emptyList() : spec.getInitContainers();
+  }
+
+  /**
+   * @param filterPredicate Predicate to filter the volumes.
+   * @return The list of filtered Volumes derived from the POD specified in the template.
+   */
+  public List<V1Volume> getVolumes(Predicate<? super V1Volume> filterPredicate) {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    if (null == spec) {
+      return Collections.emptyList();
+    }
+    List<V1Volume> volumes = spec.getVolumes();
+    return volumes == null ?
+        Collections.emptyList() :
+        volumes.stream().filter(filterPredicate).collect(Collectors.toList());
+  }
+
+  /**
+   * @return The list of filtered Volumes derived from the POD specified in the template.
+   */
+  public List<V1Volume> getVolumes() {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    return spec == null ? Collections.emptyList() : spec.getVolumes();
+  }
+
+  /**
+   * This method returns volume mounts only for the first container.
+   *
+   * @param filterPredicate Predicate to filter the Volume Mounts.
+   * @return The list of filtered Volume Mounts to the appContainer derived from the POD specified
+   * in the template.
+   */
+  public List<V1VolumeMount> getContainerVolumeMounts(
```

Comment:
This one needs a little more thought. 
There will be times when explicitly don't want to mount volumes present in an init container onto the application (non-init) container. 
For example when one of the init containers is mounting any secrets (certificates for example) that the application-container should not have access to.

One workable but little-less-than-ideal solution would be to dedicate a container in the template, identified by the container name, such that only the volume mounts present on that container will be used for merging within this method.  The name of that container can be either hardcoded or a configurable property. 
What do you think ?

Reply:
In this one, we are not getting volume mounts for init containers, but only the volume mounts for the flow container which is specified in the template.

Reply:
That makes more sense, I didn't realize the call for getting 'container' and 'init-containers' were separate.
In this case it will good enough to document the assumption that there is only one 'non-init-container' present in the template, otherwise someone trying to add more than one container might find the behavior unusual.

Reply:
done